source/source_test.go

```diff
@@ -0,0 +1,167 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/conduitio-labs/conduit-connector-clickhouse/config"
+	"github.com/conduitio-labs/conduit-connector-clickhouse/source/mock"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/golang/mock/gomock"
+	"github.com/matryer/is"
+)
+
+const (
+	url   = "http://username:password@host1:8123/database"
+	table = "test_table"
+)
+
+func TestSource_Configure(t *testing.T) {
+	is := is.New(t)
+
+	s := Source{}
+
+	err := s.Configure(context.Background(), map[string]string{
+		config.URL:            url,
+		config.Table:          table,
+		config.KeyColumns:     "id",
+		config.OrderingColumn: "created_at",
+	})
+	is.NoErr(err)
+	is.Equal(s.config, config.Source{
+		Configuration: config.Configuration{
+			URL:   url,
+			Table: table,
+		},
+		KeyColumns:     []string{"id"},
+		OrderingColumn: "created_at",
+		BatchSize:      1000,
+	})
+}
+
+func TestSource_Configure_fail(t *testing.T) {
+	is := is.New(t)
+
+	s := Source{}
+
+	err := s.Configure(context.Background(), map[string]string{
+		config.URL:        url,
+		config.Table:      table,
+		config.KeyColumns: "id",
+	})
+	is.True(err != nil)
```

Comment:
I believe it's generally useful to check not only if there is an error, but also what the error is.

Reply:
Agree with you 👍 
Updated.